app/src/main/java/com/example/codingtest1/ContactsAdapter.kt

```diff
@@ -53,16 +58,52 @@ class ContactsAdapter(var contactList: ArrayList<ContactsData>,  private val sha
         popUpMenus.setOnMenuItemClickListener {
             when (it.itemId) {
                 R.id.editText -> {
-                    Toast.makeText(view.context, "Edit text button is clicked", Toast.LENGTH_LONG)
+                    val inflater = LayoutInflater.from(view.context)
+                    val editView = inflater.inflate(R.layout.add_contact, null)
+                    val editName = editView.findViewById<EditText>(R.id.etName)
+                    val editNumber = editView.findViewById<EditText>(R.id.etNumber)
+                    val editDescription = editView.findViewById<EditText>(R.id.etDescription)
+
+                    val selectedContact = contactList[position]
+                    editName.setText(selectedContact.name)
+                    editNumber.setText(selectedContact.phoneNumber)
+                    editDescription.setText(selectedContact.description)
+
+                    AlertDialog.Builder(view.context)
+                        .setView(editView)
+                        .setPositiveButton("OK") { _, _ ->
+                            val newName = editName.text.toString()
+                            val newNumber = editNumber.text.toString()
+                            val newDescription = editDescription.text.toString()
+                            if (newName.isNotEmpty() && newNumber.isNotEmpty() && newDescription.isNotEmpty()) {
```

Comment:
You can have a separate functions for displaying the dialog and handling the logic for editing the contact.